src/internal/connector/onedrive/collections.go

```diff
@@ -433,15 +466,57 @@ func (c *Collections) UpdateCollections(
 			// already created and partially populated.
 			updatePath(newPaths, *item.GetId(), folderPath.String())
 
+			col, found := c.CollectionMap[folderPath.String()]
+			if !found {
+				// Skipping new folders here as for non empty folders
+				// we will end up creating a folder in the next case
+				// statement and we are not backing up empty folders.
+				if prevPath != nil {
+					col := NewCollection(
+						c.itemClient,
+						folderPath,
+						prevPath,
+						driveID,
+						c.service,
+						c.statusUpdater,
+						c.source,
+						c.ctrl,
+						invalidPrevDelta,
+					)
+					c.CollectionMap[folderPath.String()] = col
+					c.NumContainers++
+				}
+			} else {
+				// If we have previously created a collection when
+				// we encountered a file, this gives us a change
+				// to mark the container as moved or not moved
+				// instead of new.
+				ccol, ok := col.(*Collection)
+				if !ok {
+					logger.Ctx(ctx).Errorw("unable to cast onedrive collection", "error", err)
+					return err
+				}
+
+				ccol.SetPreviousPath(prevPath)
+				c.CollectionMap[folderPath.String()] = ccol
```

Comment:
hmmmm, there's still a potential issue here I think. I think it's still possible to end up with the collection in the map 2x (or more) if we have an order of operations like

1. start fetching delta results
2. see folder at path1
3. folder moves to path2
4. see folder at path2

Looking up the folder by the current path in our map won't suffice because the path has changed and the lookup will return nil. We may need to expand Collections to have a second map of folder ID -> *Collection? then we could do something like

```go
col, found := c.collectionsByID[*item.GetId()]
if !found {
  // Do other things like the code already has
} else {
  ccol, ok := col.(*Collection)
  if !ok {
    // log and/or return error
  }

  delete(c.CollectionMap, col.FullPath().String())
  ccol.SetPreviousPath(prevPath)
  c.CollectionMap[folderPath.String()] = ccol
}
```

If you add an additional map to `Collections` (I haven't thought of alternative approaches so you may come up with a different solution) please make a separate PR for it. We can then review and merge that PR before this one goes in

Reply:
From my testing, it does not seem like a rename in between fetching delta queries result in multiple delta objects. That said, there is "The same item may appear more than once in a delta feed, for various reasons. You should use the last occurrence you see." in the [docs](https://learn.microsoft.com/en-us/graph/api/driveitem-delta?view=graph-rest-1.0&tabs=http#remarks) which might point to a possibility of this.

My best guess is that they are sort of paging on ids on the graph backend and he skip token has that information.

Reply:
I think I have a relatively simpler solution. You can find it in [this commit](https://github.com/alcionai/corso/pull/2407/commits/0220124eddfcda8cdc8e2289a66a71988e65c763). Did not bother making a separate PR as we are not changing anything in the original Collections interface. I believe this should work.

Reply:
However any thoughts on how we would handle "The same item may appear more than once in a delta feed, for various reasons. You should use the last occurrence you see." for items?